src/sardana/macroserver/msparameter.py

```diff
@@ -395,6 +395,11 @@ def decodeNormal(self, raw_param, param_def):
                     value = raw_param.get("value")
                 else:
                     value = raw_param
+                if type(value) == list:
+                    if len(value) == 0:
+                        value = param_def['default_value']
+                    else:
+                        value = value[0]
```

Comment:
Since in #405 we agreed that the repeat parameters composed from just one member are not passed as list I would raise `WrongParam` exception here.